src/mongo/registerMongoCommands.ts

```diff
@@ -33,95 +37,68 @@ export function registerMongoCommands(): void {
+
+    // #region Collection command
+
+    registerCommandWithTreeNodeUnwrapping('cosmosDB.openCollection', openMongoCollection);
+    registerCommandWithTreeNodeUnwrapping('cosmosDB.createMongoDocument', createMongoDocument);
     registerCommandWithTreeNodeUnwrapping('cosmosDB.deleteMongoCollection', deleteMongoCollection);
-    registerCommandWithTreeNodeUnwrapping('cosmosDB.deleteMongoDocument', async (context: IActionContext, node?: MongoDocumentTreeItem) => {
-        const suppressCreateContext: ITreeItemPickerContext = context;
-        suppressCreateContext.suppressCreatePick = true;
-        if (!node) {
-            node = await pickMongo<MongoDocumentTreeItem>(context, MongoDocumentTreeItem.contextValue);
-        }
-        await node.deleteTreeItem(context);
-    });
-    registerCommandWithTreeNodeUnwrapping('cosmosDB.openCollection', async (context: IActionContext, node?: MongoCollectionTreeItem) => {
-        if (!node) {
-            node = await pickMongo<MongoCollectionTreeItem>(context, MongoCollectionTreeItem.contextValue);
-        }
-        await ext.fileSystem.showTextDocument(node);
-    });
-    registerCommandWithTreeNodeUnwrapping('cosmosDB.launchMongoShell', launchMongoShell);
-    registerCommandWithTreeNodeUnwrapping('cosmosDB.newMongoScrapbook', async () => await vscodeUtil.showNewFile('', 'Scrapbook', '.mongo'));
-    registerCommandWithTreeNodeUnwrapping('cosmosDB.executeMongoCommand', async (context: IActionContext, position?: vscode.Position) => {
-        await loadPersistedMongoDBTask;
-        await executeCommandFromActiveEditor(context, position);
-    });
-    registerCommandWithTreeNodeUnwrapping('cosmosDB.executeAllMongoCommands', async (context: IActionContext) => {
-        await loadPersistedMongoDBTask;
-        await executeAllCommandsFromActiveEditor(context);
-    });
+
+    // #endregion
+
+    // #region Document command
+
+    registerCommandWithTreeNodeUnwrapping('cosmosDB.deleteMongoDocument', deleteMongoDocument);
+
+    // #endregion
 }
 
-async function loadPersistedMongoDB(): Promise<void> {
-    // NOTE: We want to make sure this function never throws or returns a rejected promise because it gets awaited multiple times
-    await callWithTelemetryAndErrorHandling('cosmosDB.loadPersistedMongoDB', async (context: IActionContext) => {
-        context.errorHandling.suppressDisplay = true;
-        context.telemetry.properties.isActivationEvent = 'true';
-
-        try {
-            const persistedNodeId: string | undefined = ext.context.globalState.get(connectedMongoKey);
-            if (persistedNodeId) {
-                const persistedNode = await ext.rgApi.appResourceTree.findTreeItem(persistedNodeId, context);
-                if (persistedNode) {
-                    await ext.mongoLanguageClient.client.onReady();
-                    await vscode.commands.executeCommand('cosmosDB.connectMongoDB', persistedNode);
+export async function loadPersistedMongoDB(): Promise<void> {
```

Comment:
Why did you not move this one to the `loadPersistedMongoDB.ts` file?

Reply:
Sorry, I forgot this. I will move it to a separate file as well.

Reply:
Yeah, looks like you already have one but it was just empty.

Reply:
I decided to put it in the same file of connectMongoDBDatabase because loadPersistedMongoDB doesn't have its own command and is closely related to the connectMongoDBDatabase command.

I am going to keep setUpErrorReporting() and updateErrorsInScrapbook() in the registerMongoCommands.ts for now since they aren't used in any command.